tableaudocumentapi/archivefile.py

```diff
@@ -0,0 +1,67 @@
+import contextlib
+import os
+import shutil
+import tempfile
+import zipfile
+
+import xml.etree.ElementTree as ET
+
+
+@contextlib.contextmanager
+def temporary_directory(*args, **kwargs):
+    d = tempfile.mkdtemp(*args, **kwargs)
+    try:
+        yield d
+    finally:
+        shutil.rmtree(d)
+
+
+def find_file_in_zip(zip, ext):
+    for filename in zip.namelist():
+        if os.path.splitext(filename)[-1].lower() == ext[:-1]:
```

Comment:
Good call, I'll look into it, along with your later suggestion of when to extract vs open directly.

Originally my brain didn't want to wrap itself around how to deal with the read-only stream and writing to files, but now that I've had coffee...